test/test_point_inclusion.cpp

```diff
@@ -359,9 +366,17 @@ TEST(CylinderPointContainment, Basic)
     EXPECT_FALSE(cylinder.containsPoint(Eigen::Vector3d(0.00, 0.00, 2.01)));
 
     // near two-axis maximum
+    // HACK: sqrt(2) / 2 is a problem due to rounding errors since the distance
+    //       is computed as 1.0000000000000002 . In such case we subtract an
+    //       epsilon. For the case when 1+sq2 is used, this does not occur.
     const double sq2 = sqrt(2) / 2;
+    double sq2_eps = sq2;
+    if (sq2_eps * sq2_eps + sq2_eps * sq2_eps > 1)
+      sq2_eps -= std::numeric_limits<double>::epsilon();
+    ASSERT_LE(sq2_eps * sq2_eps + sq2_eps * sq2_eps, 1.0);
+
     EXPECT_TRUE( cylinder.containsPoint(Eigen::Vector3d(0.70, 0.70, 0.00)));
-    EXPECT_SURF( cylinder.containsPoint(Eigen::Vector3d(sq2 , sq2 , 0.00)));
+    EXPECT_SURF(cylinder.containsPoint(Eigen::Vector3d(sq2_eps , sq2_eps , 0)));
```

Comment:
Please revert these changes. They shouldn't be necessary anymore with `EXPECT_DOUBLE_EQ`.